### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by fllaca**
Co-authored-by: Marc de Klerk marc@adhara.io

### CODE REVIEW

**Review by hrodes — COMMENTED** [review]
Please add a license file

**Comment by marcdk** [comment]
just a few notes:
- tested with https://github.com/AdharaProjects/cluster-config-apps/pull/1171
- Apache 2.0 licence added
- changelog added
- only a publish job added to the pipeline since no tests yet

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	@echo "Building in docker ${DOCKER_IMAGE_NAME}:${DOCKER_IMAGE_TAG}"
+	docker build -t ${DOCKER_IMAGE_NAME}:${DOCKER_IMAGE_TAG} --build-arg GOPROXY=${GOPROXY} -t ${DOCKER_IMAGE_NAME}:${DOCKER_IMAGE_TAG} .
+
+.PHONY: publish
+publish: build_in_docker
+	docker push ${DOCKER_IMAGE_NAME}:${DOCKER_IMAGE_TAG}
+
+.PHONY: sonar-scanner
+sonar-scanner:
+	scripts/sonar.sh adhara-crypto
```
Comment (figuerascarlos) [inline Makefile:74]: this is OK ?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+validate: check_env
+
+.PHONY: test
+test: validate
+	@echo "Placeholder Makefile rule for tests"
+
+.PHONY: build_in_docker
+build_in_docker: check_docker_env
+	@echo "Building in docker ${DOCKER_IMAGE_NAME}:${DOCKER_IMAGE_TAG}"
+	docker build -t ${DOCKER_IMAGE_NAME}:${DOCKER_IMAGE_TAG} --build-arg GOPROXY=${GOPROXY} -t ${DOCKER_IMAGE_NAME}:${DOCKER_IMAGE_TAG} .
```
Comment (figuerascarlos) [inline Makefile:66]:
```suggestion
	docker build -t ${DOCKER_IMAGE_NAME}:${DOCKER_IMAGE_TAG}  -t ${DOCKER_IMAGE_NAME}:${DOCKER_IMAGE_TAG} .
```

File: cmds/connectnodes.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
+package cmds
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"k8s.io/client-go/rest"
+
+	"github.com/adharaprojects/kubesu/pkg/rpc"
```
[the remark this answers is not in the document]
Comment (figuerascarlos) [reply cmds/connectnodes.go]: I think so yes

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Changelog
+
+## 0.0.3
```
Comment (fllaca) [inline CHANGELOG.md:3]: I think we don't need 0.0.1-2-3, as this is the very first PR to master. I would collapse everything in `[unreleased]` or just `0.0.1` . Also please add some disclaimer saying that this is a first prototype version that is not intended for production yet (both in CHANGELOG and README).

File: cmds/connectnodes.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (fllaca) [reply cmds/connectnodes.go]: +1

**Comment by marcdk** [comment]
> I would remove the config for besu referencing a private chart in examples folder.

I've removed the entire examples folder, if that's good for now? Then reworked examples can be added in a follow-up PR, else I can add reworked examples now

File: go.mod [hunk 5]
```diff
@@ -0,0 +1,18 @@
+module github.com/adharaprojects/kubesu
```
[the remark this answers is not in the document]
Comment (marcdk) [reply go.mod:1]: entire examples/ folder removed https://github.com/Adhara-Tech/kubesu/pull/1#issuecomment-800319715

File: Makefile (hunk 1 above)
Comment (marcdk) [reply Makefile]: Nope, not OK :) Removed in latest commit

**Comment by marcdk** [comment]
commits will squashed and `master` tagged with `0.0.1` once merged

File: CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# Changelog
+
+## 0.0.1
```
Comment (fllaca) [inline CHANGELOG.md:3]:
```suggestion
## 0.0.1-rc1
```
? to reinforce its "beta" status

**Review by fllaca — COMMENTED** [review]
just one final comment on the first version name. Can you also please squash everything in a single commit? It can be done on Github UI when merging, but as I'm author of the PR I cannot do it (nor approve this xD)

File: CHANGELOG.md (hunk 6 above)
Comment (marcdk) [reply CHANGELOG.md:3]: 100% and I'll use the github cli ;)

**Pull request closed without merging**